go/vt/vttablet/tabletserver/debugenv.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2020 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tabletserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"html"
+	"net/http"
+	"strconv"
+	"text/template"
+
+	"vitess.io/vitess/go/acl"
+	"vitess.io/vitess/go/vt/log"
+)
+
+var (
+	debugEnvHeader = []byte(`
+	<thead><tr>
+		<th>Variable Name</th>
+		<th>Value</th>
+		<th>Action</th>
+	</tr></thead>
+	`)
+	debugEnvRow = template.Must(template.New("debugenv").Parse(`
+	<tr><form method="POST">
+		<td>{{.VarName}}</td>
+		<td>
+			<input type="hidden" name="varname" value="{{.VarName}}"></input>
+			<input type="text" name="value" value="{{.Value}}"></input>
+		</td>
+		<td><input type="submit" name="Action" value="Modify"></input></td>
+	</form></tr>
+	`))
+)
+
+type envValue struct {
+	VarName string
+	Value   string
+}
+
+func debugEnvHandler(tsv *TabletServer, w http.ResponseWriter, r *http.Request) {
```

Comment:
Probably we should reject the request if the method is not **POST**

Reply:
I actually rely on this for my tests, by using GET. Is there a reason to disallow? Also, we seem to be allowing it in other places.

Reply:
I saw that, If you change this, you will have to update the tests. 

I think the reason is general HTTP conventions. In theory, `GET` requests could be cached and shouldn't mutate state.

I think that ideally, we should stop propagating this pattern. Once people start relying on the get will become even more difficult to change it.